utils_token.go

```diff
@@ -0,0 +1,132 @@
+//Because Onedrive only use OAuth2.0
+
+//https://docs.microsoft.com/en-us/graph/auth/auth-concepts
+
+//we use `code flow` to get the oauth2 access_token
+
+//https://docs.microsoft.com/en-us/advertising/guides/authentication-oauth-get-tokens?view=bingads-13
+package onedrive
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"golang.org/x/oauth2"
+)
+
+const (
+	//First, we need to access ` get_ code_ URL ` get the `code` from the attached information in the redirected URL
+	get_code_url  = "https://login.live.com/oauth20_authorize.srf?response_type=code&redirect_uri=http://localhost:9999&response_mode=query&scope=offline_access%20user.read%20mail.read%20Files.ReadWrite.All&client_id=8ce1780f-12cc-45eb-a4c9-83f284e56621"
+	get_token_url = "https://login.live.com/oauth20_token.srf"
+)
+
+//for test we can use the client I have registered
+const (
+	secret       = "-uGEL0oQZ86Wwr2I-dTF3~Y3i6sCGE5z~n"
+	client_id    = "8ce1780f-12cc-45eb-a4c9-83f284e56621"
+	redirect_url = "http://localhost:9999"
+)
```

Comment:
Generally these sensitive data should not be pushed to GitHub. (You can use GitHub secrets when necessary.) And it seems that we don't need `utils_token` as https://github.com/beyondstorage/go-service-onedrive/pull/12#issuecomment-898895801

Reply:
You can try to remove this file from commit history... (a chance to learn advanced git operation 🤣